### ISSUE TITLE
spark.end(data) does not send data in some cases

### ISSUE THREAD

**Issue by wizcorp-olivier** [primus/primus#395]
In the same configuration (same line of code of the same process at a few seconds interval), we saw ```spark.end(data)``` actually send the data before closing, but in some cases it did not send - spark was closed but data was not received on the other side.

Versions used:
"primus": "3.2.1"
"engine.io": "1.5.2"
"iojs": "3.0.0"

**Comment by 3rd-Eden** [comment]
Could it be that this browser, device or even network specific? Could you give us some more information on how this was tested?

**Comment by wizcorp-olivier** [comment]
It showed the same issue with Chrome on Mac, UiWebview on iOS8 and Crosswalk on Android.
The Node server was running on a Mac.

One example:
- server calls spark.end(data)
- client receives data, then spark is closed (as expected)
- client opens a new connection

A few seconds or less after:
- server calls spark.end(data)
- data is not received but client's spark gets closed

**Comment by wizcorp-olivier** [comment]
Any hint or further question?

**Comment by 3rd-Eden** [comment]
No, it seems to be a bug then which is most likely caused by a race condition somewhere. I wonder if this is something that is specific to a framework but it would still be something that we need to resolve.
